### PR TITLE
Log 10569 profiling target

### PR DESCRIPTION
**Pull request by c-nixon**
This adds a benchmark as well as a make bench target to run it. When run it will drop file named flamegraph.svg in the repo root directory

### CODE REVIEW

File: bench/bin/throughput.rs [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+    rpb.set_style(sty.clone());
+    rpb.set_message("Received:    ");
+    rpb.tick();
+
+    let mp = tokio::task::spawn_blocking({
+        let mp = m.clone();
+        move || mp.join().unwrap()
+    });
+
+    //println!("ticked bars");
```
Comment (dhable) [inline bench/bin/throughput.rs:149]: nitpick:  This could be removed

File: bench/bin/throughput.rs [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+                            std::thread::sleep(std::time::Duration::from_millis(50));
+                            wpb.inc(10_000);
+                            if count % 100_000 == 0 {
+                                log.flush().unwrap();
+                            }
+                        }
+                        writeln!(log, "{}", word).unwrap();
+                    }
+
+                    //println!("Written lines: {}", count);
```
Comment (dhable) [inline bench/bin/throughput.rs:260]: nit: also could be removed

File: bench/bin/throughput.rs [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+                            wpb.inc(10_000);
+                            std::thread::sleep(std::time::Duration::from_millis(50));
+                            if count % 100_000 == 0 {
+                                log.flush().unwrap();
+                            }
+                        }
+                        writeln!(log, "{}", word).unwrap();
+                    }
+
+                    //println!("Written lines: {}", count);
```
Comment (dhable) [inline bench/bin/throughput.rs:276]: nit: and this one as well

**Review by dhable — COMMENTED** [review]
Looks good. Just a few nitpicks about commented out code that could be removed.

**Comment by c-nixon** [comment]
I have one last change to make to this pending the change to build images. Once that's been merged I'll be able to remove the Dockerfile.bench from this to avoid duplicating base image Dockerfile in here

**Review by dhable — APPROVED** [review]

**Pull request merged**